lib/pbench/cli/agent/commands/tools/list.py

```diff
@@ -0,0 +1,136 @@
+"""
+pbench-list-tools
+
+This script can list all tools from all groups, list tools from a specific
+group, or list which groups contain a specific tool
+
+"""
+
+import sys
+
+import click
+
+from pbench.cli.agent import CliContext, pass_cli_context
+from pbench.cli.agent.commands.tools.base import ToolCommand
+from pbench.cli.agent.options import common_options
+
+
+class ListTools(ToolCommand):
+    """ List registered Tools """
+
+    def __init__(self, context):
+        super(ListTools, self).__init__(context)
+
+    def execute(self):
+        if not self.pbench_run.exists():
+            self.logger.warn("The %s directory does not exist", self.pbench_run)
```

Comment:
I believe the logging format in the agent code has to use `{}` style now.  Can you confirm with a unit test?

Reply:
Still needed ...